e2e-tests/support/trade-finance-manager/tfmLogin.js

```diff
@@ -0,0 +1,104 @@
+const crypto = require('crypto');
+const jws = require('jws');
+const cookieSignature = require('cookie-signature');
+const { TFM_URL } = require('../../e2e-fixtures/constants.fixture');
+
+const issueValidJWT = (user, sessionIdentifier) => {
+  const { _id } = user;
+  const payload = {
+    sub: _id,
+    username: user.username,
+    teams: user.teams,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    sessionIdentifier,
+  };
+  const secret = Buffer.from(Cypress.config('jwtSigningKey'), 'base64').toString('ascii');
+
+  const header = {
+    alg: 'RS256',
+    typ: 'JWT',
+  };
+  return jws.sign({
+    header,
+    payload,
+    secret,
+    encoding: 'utf8',
+  });
+};
+
+/**
+ * tfmLogin
+ * Ensure a user is logged in to TFM.
+ * Return JWT token for API use.
+ * 1) Create season id
+ * 2) Set session id in MondoDB
+ * 3) Set session in Redis
+ * 4) Set session id in Cookie, if isSessionForAPICall is false (default behaviour)
+ * 5) Redirects to TFM root /, if isSessionForAPICall is false (default behaviour)
+ * @param {Object} user: User object for login
+ * @param {Boolean} isSessionForAPICall: if true no cookie set and no redirect will happen.
+ * @returns {Promise<String>} TFM JWT token.
+ *
+ * @example
+ * // Returns "Bearer ..."
+ * cy.tfmLogin({user});
+ *
+ * @example
+ * // Returns "Bearer ..."
+ * cy.tfmLogin({user, isSessionForAPICall: true}).then((token) => apiCall(token));
+ *
+ */
+const tfmLogin = ({
+  user,
+  isSessionForAPICall = false,
+}) => {
+  const { username } = user;
+
+  // TODO: try reuse login session.
```

Comment:
I'm not sure what is meant here by reusing a login session, at this stage we're not logged in?

Reply:
I wanted to discus tracking of active user and skip login if new user is same as current.

But overhead is not high, extra sessions in Redis doesn't take much space and will auto expire in 30 minutes.

Now I would vote that we can always login without checking current session.